Formula/libvisual-projectm.rb

```diff
@@ -0,0 +1,42 @@
+class LibvisualProjectm < Formula
+  desc "Visualization plug-in for projectM support from Libvisual"
+  homepage "https://github.com/projectM-visualizer/frontend-libvisual-plug-in"
+  url "https://github.com/projectM-visualizer/frontend-libvisual-plug-in/archive/refs/tags/v2.1.1.tar.gz"
+  sha256 "eb8269c2a923546600d3f40ff90c011f03a215847215ee8bef44bfae305b4dd7"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "libvisual"
+  depends_on "projectm"
+
+  def install
+    # We cannot write to libvisual's cellar, so we need to deflect
+    # installation and ask the user to create a symlink via caveats.
+    inreplace "CMakeLists.txt",
+      "LIBRARY DESTINATION \"${LIBVISUAL_ACTOR_PLUGINS_DIR}\"",
+      "LIBRARY DESTINATION \"#{lib}\""
```

Comment:
@carlocab from what I have seen in CI in the past, I believe we will not be able to install plugins there in CI and also not be able to install symlinks to files installed elsewhere from post_install. CI seems to say "no" to everything but the small box that the package has. As long as the CI is what it is today, we can probably save the time on that to only see that fail again. I'm happy to apply sure fixes but my battery for experiments is near empty here after many hours of CI saying no to things that worked fine outside of CI, if I'm honest. How would you like to continue?

Reply:
Ah, sorry, what I meant above was actually:
```
#define VISUAL_PLUGIN_PATH "#{HOMEBREW_PREFIX}/lib/libvisual-#{libvisual.version.major_minor}"
```

While packages get their own private prefix
- `libvisual` gets `$HOMEBREW_PREFIX/Cellar/libvisual/0.4.2`
- `libvisual-plugins` gets `$HOMEBREW_PREFIX/Cellar/libvisual-plugins/0.4.2`
and we have to make them install things _only_ into their own prefix, `brew` also creates a symlinks
```
$HOMEBREW_PREFIX/Cellar/libvisual-plugins/0.4.2/lib/* -> $HOMEBREW_PREFIX/lib
```
so if `libvisual` looks for plugins in, say,
```
$HOMEBREW_PREFIX/lib/libvisual-0.4.2/actor
```
it will find things installed by `libvisual-plugins`, because they've been symlinked into that location.

Does that make sense?

Reply:
@carlocab (prior to any changes) I caught this in output of `brew install libvisual`:

```
==> Finishing up
[..]
ln -s ../Cellar/libvisual/0.4.2/lib/libvisual-0.4 libvisual-0.4
[..]
```

If I read that right, `lib/libvisual-0.4` is wired to package `libvisual` via symlink in its entirety. We would need a way to keep brew from doing that, for your idea to work for subfolder `actor`. What do you think?

Reply:
`brew` should turn that into a real directory if another formula wants to install something in `lib/libvisual-0.4`.

Reply:
@carlocab do we have a deal that if CI says ends up saying "no! permissions!" we revert and stick to the approach we have now?

Reply:
@carlocab moment of truth, pushed, let's see what the CI says…